device/celestica/x86_64-cel_silverstone-r0/plugins/sfputil.py

```diff
@@ -379,9 +584,34 @@ def get_eeprom_dict(self, port_num):
                 sfp_data['interface'] = sfpi_obj.get_data_pretty()
 
                 # check if it is a 100G module
-                if sfp_data['interface']['data']['Identifier'] == 'QSFP28 or later':
+                if sfp_data['interface']['data']['Identifier'] not in [type_of_transceiver['18'], type_of_transceiver['19']]:
                     return self.get_qsfp_data(eeprom_ifraw)
 
+                # decode application advertisement
+                offset = 85
+                tbl = self.parse_media_type(eeprom_ifraw, offset)
+                ret = ""
+                if tbl is not None:
+                    app = 1
+                    hid = int(eeprom_ifraw[1 + offset], 16)
+                    while (hid != 0) and (hid != 0xff):
```

Comment:
1. This is to decode the application advertisement and get it displayed on the console.
e.g.
admin@sonic:~$ show interfaces transceiver eeprom
Ethernet0: SFP EEPROM detected
        Application Advertisement:
            1: 400GAUI-8 C2M (Annex 120E) | 400GBASE-DR4 (Cl 124)
            2: 100GAUI-2 C2M (Annex 135G) | 100GBASE-DR (Cl 140)
        Application Selected: 1
        Identifier: QSFP-DD Double Density 8X Pluggable Transceiver
        Interrupt Asserted: Off
        Module State: Ready State
        Revision Compliance: 3.0
        Vendor Name: Eoptolink
        Vendor Part Number: EOLD-134HG-02-M6
        Vendor Revision: 01
        Vendor Serial Number: DKC9A90005

2.  Yes, it could be an infinity loop if the number of supported application mode >=8, will add a sanity check for this.